src/routes/api/auth.js

```diff
@@ -0,0 +1,33 @@
+const KoaRouter = require('koa-router');
+const jwtgenerator = require('jsonwebtoken');
+require('dotenv').config();
+
+const router = new KoaRouter();
+
+async function generateToken(user) {
+  return new Promise((resolve, reject) => {
+    jwtgenerator.sign(
+      { userId: user.id },
+      process.env.JWT_SECRET,
+      (err, tokenResult) => (err ? reject(err) : resolve(tokenResult)),
+    );
+  });
+}
+
+router.post('api.auth.login', '/', async (ctx) => {
+  const { email, password } = ctx.request.body;
+  const user = await ctx.orm.user.findOne({ where: { email } });
+  if (!user) {
+    ctx.throw(404, `No user found with ${email}`);
+  } else if (!await user.checkPassword(password)) {
+    ctx.throw(401, 'Invalid password');
+  }
```

Comment:
Una sugerencia que creo que se ve algo más simple de leer. Si decides aceptarla, verifica que no levante errores de ESLint.
```suggestion
  if (!user) ctx.throw(404, `No user found with ${email}`);
  
  const authenticated = await user.checkPassword(password);
  if (!authenticated) ctx.throw(401, 'Invalid password');
```

Reply:
Al final eslint si acepta one liners :smiley: